modules/gapi/src/backends/ie/giebackend.hpp

```diff
@@ -37,6 +38,37 @@ struct IECompiled {
     InferenceEngine::InferRequest      this_request;
 };
 
+// FIXME: Structure which collect all necessary sync primitives
+// will be deleted when the async request pool appears
+class SyncPrim {
+public:
+    void wait() {
+        std::unique_lock<std::mutex> l(m_mutex);
```

Comment:
so if you do this `wait()` at the early beginning, I don't get why don't you get stuck immediately in a deadlock.

Reply:
Because condition (`is_busy`) is false ?

Reply:
Yes I see now